core/src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala

```diff
@@ -104,6 +100,52 @@ sealed abstract class ChainParams {
                        amount)
   }
 
+  /**
+   * Creates the Genesis [[org.bitcoins.core.protocol.blockchain.Block Block]] for Signet.
+   *
+   * @see Mimics
+   *      [[https://github.com/bitcoin/bitcoin/blob/006d9ab9e59a00fa02b8d5f32919f9d57567fa65/src/chainparams.cpp#L67 this function]]
+   *      in Bitcoin Core
+   *
+   * @param time the time when the miner started hashing the block header
+   * @param nonce the nonce to mine the block
+   * @param nBits An encoded version of the target threshold this block’s header hash must be less than or equal to.
+   * @param version the block version
+   * @param amount the block reward for the genesis block (50 BTC in Bitcoin)
+   * @return the newly minted genesis block
+   */
+  def createSignetGenesisBlock(
+      time: UInt32,
+      nonce: UInt32,
+      nBits: UInt32,
+      version: Int32,
+      amount: CurrencyUnit): Block = {
+    val defaultSignetKey = DoubleSha256Digest(
+      "03ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430")
+    val const = ScriptConstant(defaultSignetKey.bytes)
+
+    val asm = {
+      List(OP_1, const, OP_1, OP_CHECKMULTISIG)
+    }
+
+    val blockScript = ScriptSignature.fromAsm(asm)
+    val genesisOutputScript = ScriptPubKey.fromAsm(Seq(OP_RETURN))
+
+    val input = CoinbaseInput(blockScript, TransactionConstants.sequence)
+    val output = TransactionOutput(amount, genesisOutputScript)
+    val tx = BaseTransaction(TransactionConstants.version,
+      Seq(input),
+      Seq(output),
+      TransactionConstants.lockTime)
+    val prevBlockHash = DoubleSha256Digest(
+      "0000000000000000000000000000000000000000000000000000000000000000")
+    val merkleRootHash = Merkle.computeMerkleRoot(Seq(tx))
+    val genesisBlockHeader =
+      BlockHeader(version, prevBlockHash, merkleRootHash, time, nBits, nonce)
+    val genesisBlock = Block(genesisBlockHeader, Seq(tx))
+    genesisBlock
```

Comment:
nit: you could not define this as a `val` and just have `Block(genesisBlockHeader, Seq(tx))`